docs/api/top-level-window.md

```diff
@@ -0,0 +1,1240 @@
+# TopLevelWindow
+
+> Create and control native windows.
+
+Process: [Main](../glossary.md#main-process)
+
+## Frameless window
+
+To create a window without chrome, or a transparent window in arbitrary shape,
+you can use the [Frameless Window](frameless-window.md) API.
+
+## Parent and child windows
+
+By using `parent` option, you can create child windows:
+
+```javascript
+const { TopLevelWindow } = require('electron')
+
+let top = new TopLevelWindow()
+let child = new TopLevelWindow({ parent: top })
+child.show()
+top.show()
+```
+
+The `child` window will always show on top of the `top` window.
+
+## Modal windows
+
+A modal window is a child window that disables parent window, to create a modal
+window, you have to set both `parent` and `modal` options:
+
+```javascript
+let child = new TopLevelWindow({ parent: top, modal: true, show: false })
+child.show()
+```
+
+## Platform notices
+
+* On macOS modal windows will be displayed as sheets attached to the parent window.
+* On macOS the child windows will keep the relative position to parent window
+  when parent window moves, while on Windows and Linux child windows will not
+  move.
+* On Linux the type of modal windows will be changed to `dialog`.
+* On Linux many desktop environments do not support hiding a modal window.
+
+## Class: TopLevelWindow
+
+> Create and control native windows.
+
+Process: [Main](../glossary.md#main-process)
```

Comment:
It's odd that this information is repeated..?

Reply:
The original documentation does so, I think it is because one is for the module and one is for the class? I don't know if the docs parser needs 2 of this. @MarshallOfSound